src/transport_portals4.h

```diff
@@ -269,6 +274,18 @@ static inline void shmem_transport_probe(void) {
     return;
 }
 
+static inline
+shmem_transport_addr_t shmem_transport_get_local_addr(void)
+{
+    shmem_transport_addr_t addr;
+    int ret = PtlGetPhysId(shmem_transport_portals4_ni_h, &addr.ptl_addr);
+    if (PTL_OK != ret) {
+        RAISE_ERROR_MSG("PtlGetPhysId failed: %d\n", ret);
+    }
+    return addr;
+}
```

Comment:
The physical ID isn't quite enough to determine which processes share a node.  You need to look at the `nid` in:
```
typedef union {
  struct {
    ptl_nid_t nid;
    ptl_pid_t pid;
  } phys;
  ptl_rank_t rank;
} ptl_process_t;
```
However, you do want to store the whole `ptl_process_t` to avoid putting duplicate info into the KVS.  I think this can be remedied by adding a `shmem_transport_same_node(const shmem_transport_addr_t *a1, const shmem_transport_addr_t *a2)` comparison operation to the transport interface.

Reply:
See line 558 in transport_portals4.c for the nid usage.